src/FffGcodeWriter.cpp

```diff
@@ -1660,48 +1732,80 @@ bool FffGcodeWriter::addSupportInfillToGCode(const SliceDataStorage& storage, La
     EFillMethod support_pattern = infill_extr.getSettingAsFillMethod("support_pattern"); // first layer pattern must be same as other layers
     if (layer_nr <= 0 && (support_pattern == EFillMethod::LINES || support_pattern == EFillMethod::ZIG_ZAG)) { support_pattern = EFillMethod::GRID; }
 
-    int infill_extruder_nr_here = (layer_nr <= 0)? getSettingAsIndex("support_extruder_nr_layer_0") : getSettingAsIndex("support_infill_extruder_nr");
-    const ExtruderTrain& infill_extr_here = *storage.meshgroup->getExtruderTrain(infill_extruder_nr_here);
-
-    const Polygons& support = support_layer.supportAreas;
-
-    std::vector<PolygonsPart> support_islands = support.splitIntoParts();
+    int infill_extruder_nr_here = (layer_nr <= 0) ? getSettingAsIndex("support_extruder_nr_layer_0") : getSettingAsIndex("support_infill_extruder_nr");
 
+    // create a list of outlines and use PathOrderOptimizer to optimize the travel move
+    std::vector<Polygons> infill_outline_list;
+    for (unsigned int part_idx = 0; part_idx < support_layer.support_infill_parts.size(); ++part_idx)
+    {
+        const bool has_wall = !support_layer.support_infill_parts[part_idx].insets.empty();
```

Comment:
I agree, and this means all parts will have their own outline. Because the `supportAreas` seems difficult to get rid of at the moment, those outlines will be redundant to the `supportAreas`. If you think this is ok, I can add the outlines to the parts and use that instead.